### PR TITLE
Adding crocoddyl

### PR DESCRIPTION
**Pull request by jcarpent**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jcarpent** [comment]
@conda-forge/help-c-cpp this package is ready to merge. @wolfv Could you review and merge this PR? Thanks in advance for your help.

**Comment by jcarpent** [comment]
@wolfv Would you have time to handle it?

**Comment by jcarpent** [comment]
@conda-forge/help-c-cpp this package is ready to be reviewed and merged. Thanks in advance for your time.

**Comment by jcarpent** [comment]
Hi @conda-forge/help-c-cpp, @conda-forge/help-python, this recipe is ready for review.

File: recipes/crocoddyl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  number: 0
+  skip: True  # [win]
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - clang  # [win]
```
Comment (dopplershift) [inline recipes/crocoddyl/meta.yaml:22]: Does this not build with visual studio?

File: recipes/crocoddyl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - make
+  host:
+    - eigen
+    - eigenpy
+    - pinocchio
+    - boost
+    - python
+    - example-robot-data
```
Comment (dopplershift) [inline recipes/crocoddyl/meta.yaml:30]: Is the example data really needed at build time?

File: recipes/crocoddyl/meta.yaml (hunk 1 above)
Comment (jcarpent) [reply recipes/crocoddyl/meta.yaml]: Fixed.

File: recipes/crocoddyl/meta.yaml (hunk 2 above)
Comment (jcarpent) [reply recipes/crocoddyl/meta.yaml]: Fixed.

**Comment by jcarpent** [comment]
@dopplershift Thanks a lot for your comments. I addressed all of them.

File: recipes/crocoddyl/meta.yaml (hunk 2 above)
Comment (dopplershift) [reply recipes/crocoddyl/meta.yaml]: It seems, based on the failed builds, that `example-robot-data` really *is* needed at build time, so probably need to add that back.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**